### PR TITLE
WT-12017 Update truncate section of the architecture guide

### PR DESCRIPTION
**Pull request by jiechenbo**
This ticket aims to update any unclear sections in the architecture. There are a few places in the architecture guide that I have attempted to make more clear. I have also updated the information to follow the fix of WT-11845

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/docs/arch-fast-truncate.dox [hunk 1]
```diff
@@ -32,8 +32,8 @@ readers.
 The \c page_del field of a \c WT_REF, if not \c NULL, contains information about the
 transaction that deleted the page.
 Its type is \c WT_PAGE_DELETED.
-It can be thought of as a special kind of update, and contains roughly the same
-information.
+The \c page_del can be thought of a special kind of update, as they contain roughly the same
```
Comment (ruby-oujo) [inline src/docs/arch-fast-truncate.dox:35]: - you removed "as" before "a special kind" here I think on accident. 
- NP optional: "exact" here is redundant as you already have "same"

File: src/docs/arch-fast-truncate.dox [hunk 2]
```diff
@@ -64,32 +64,22 @@ Apart from instantiated pages, whose state is \c WT_REF_MEM (see @ref
 ft_instantiated_pages below) the \c page_del field will be \c NULL.
 
 Because locking the ref is expensive, and visibility checks are expensive as well, we want
```
Comment (ruby-oujo) [inline src/docs/arch-fast-truncate.dox:66]: I would rephrase this as "Locking the ref and visibility checks are expensive, we want to avoid doing either unnecessarily."

**Review by ruby-oujo — APPROVED** [review]
LGTM - minor comments.

File: src/docs/arch-fast-truncate.dox [hunk 3]
```diff
@@ -64,32 +64,22 @@ Apart from instantiated pages, whose state is \c WT_REF_MEM (see @ref
...
-searches that lead to its portion of the namespace will still land on it.
-(This triggers instantiation.)
-At some point after the deletion becomes globally visible, the \c WT_REF is discarded.
-This happens either during an internal page split or reverse split, or when the internal
-page is evicted.
-It is this discard process (rather than the deletion itself) that creates gaps in the
-namespace.
+For the visibility of deleted pages, they still "exist" in the sense that the \c WT_REF remains, and
+any searches within the btree that lead to its portion of the namespace will still land on it. In
+this case we trigger page instantiation. See @ref ft_instantiated.
```
Comment (jeremythorp) [inline src/docs/arch-fast-truncate.dox:81]: s_all is generating an error, so I think that should be ft_instantiated_pages.

**Review by jeremythorp — APPROVED** [review]
Once the reference is fixed, LGTM!

**Comment by jiechenbo** [comment]
evergreen merge
